### PR TITLE
Support User: Highlight Masterbar when active

### PR DESCRIPTION
**Pull request by jordwest**
When activating support user mode, the Masterbar is highlighted to remind you that you're not in a normal mode and could make changes that you didn't mean to.

### Normal
support_user_normal

### Support user active
support_user_active

### CODE REVIEW

File: client/state/support/reducer.js [hunk 1]
```diff
@@ -9,6 +9,16 @@ import {
 	SUPPORT_USER_RESTORE,
 } from 'state/action-types';
 
+export function isSupportUser( state = false, action ) {
```
Comment (mtias) [inline client/state/support/reducer.js]: Should this be a selector instead?

File: client/layout/index.jsx [hunk 2]
```diff
@@ -94,7 +94,8 @@ Layout = React.createClass( {
 			<MasterbarLoggedIn
 				user={ this.props.user }
 				section={ this.props.section }
-				sites={ this.props.sites } />
+				sites={ this.props.sites }
+				isSupportUser={ config.isEnabled( 'support-user' ) && this.props.isSupportUser } />
```
Comment (mtias) [inline client/layout/index.jsx]: I wouldn't pass this as a prop. I'd set a class on `layout` and use that as `.layout .masterbar {}`

File: client/layout/index.jsx [hunk 3]
```diff
@@ -143,8 +144,10 @@ Layout = React.createClass( {
 export default connect(
 	( state ) => {
 		const { isLoading, section, hasSidebar, chunkName } = state.ui;
+		const isSupportUser = state.support.isSupportUser;
```
Comment (mtias) [inline client/layout/index.jsx]: Looking at this here, it should probably be a selector: `isSupportUser( state )`

**Comment by folletto** [comment]
It's in-progress, but just a drive by with a couple of notes on the current state:

1. The shadow of the masterbar should match
2. The "New" icon should turn orange too

**Comment by dllh** [comment]
I tested this a bit and it worked well in terms of highlighting the masterbar. I did see some js errors and some rendering weirdness, but we think that's unrelated to this set of changes, and we're going to track it separately. If we can get the other feedback addressed and ping for another review once fixes are in, I'd love to get this landed in short order.

File: client/layout/index.jsx (hunk 3 above)
Comment (jordwest) [inline client/layout/index.jsx]: Done - see [6d1ee3f](https://github.com/Automattic/wp-calypso/commit/6d1ee3f24ace4d0a46182ca0c6ac0fba994dc541)

File: client/layout/index.jsx (hunk 2 above)
Comment (jordwest) [inline client/layout/index.jsx]: I've added the `is-support-user` class in [3fabde3](https://github.com/Automattic/wp-calypso/commit/3fabde3c609970b8a7db99cdf9b90711646a1c7f)

**Comment by jordwest** [comment]
@folletto: Good points, I've made those changes in [95cec5e](https://github.com/Automattic/wp-calypso/commit/95cec5e342569c192573c3279743d0aa941052af):
screen shot 2016-02-03 at 2 58 49 pm

File: client/layout/index.jsx [hunk 4]
```diff
@@ -145,6 +152,7 @@ export default connect(
 		const { isLoading, section, hasSidebar, chunkName } = state.ui;
 		return { 
 			isLoading,
+			isSupportUser: isSupportUser( state ),
```
Comment (mtias) [inline client/layout/index.jsx]: This looks much better to me, thanks.

**Comment by gwwar** [comment]
:ship: tested on Chrome, FF, Safari.

**Pull request merged**